hardware_exporter.go

```diff
@@ -0,0 +1,69 @@
+package main
+
+import (
+	"net/http"
+	"os"
+	"os/user"
+
+	"github.com/go-kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/prometheus/common/promlog"
+	"github.com/prometheus/common/promlog/flag"
+	"github.com/prometheus/common/version"
+	"github.com/prometheus/exporter-toolkit/web"
+	kingpin "gopkg.in/alecthomas/kingpin.v2"
+
+	"hardware_exporter/collector"
+)
+
+var (
+	listenAddress = kingpin.Flag(
+		"web.listen-address",
+		"Address to listen on for web interface and telemetry.",
+	).Default(":9696").String()
+	metricsPath = kingpin.Flag(
+		"web.telemetry-path",
+		"Path under which to expose metrics.",
+	).Default("/metrics").String()
+	hostsConfigFile = kingpin.Flag(
+		"config.ini",
+		"Path to get hosts yaml file that can get all server bmc info.",
+	).Default("./config.ini").String()
+	configFile = kingpin.Flag(
+		"web.config",
+		"[EXPERIMENTAL] Path to config yaml file that can enable TLS or authentication.",
+	).Default("").String()
+)
+
+func main() {
+	promlogConfig := &promlog.Config{}
+	flag.AddFlags(kingpin.CommandLine, promlogConfig)
+	kingpin.Version(version.Print("hardware_exporter"))
+	kingpin.CommandLine.UsageWriter(os.Stdout)
+	kingpin.HelpFlag.Short('h')
+	kingpin.Parse()
+	logger := promlog.New(promlogConfig)
+	prometheus.MustRegister(collector.NewRedfishCollector(*hostsConfigFile, logger))
+	level.Info(logger).Log("msg", "Starting hardware_exporter", "version", version.Info())
+	level.Info(logger).Log("msg", "Build context", "build_context", version.BuildContext())
+	if user, err := user.Current(); err == nil && user.Uid == "0" {
+		level.Warn(logger).Log("msg", "Hardware Exporter is running as root user. This exporter is designed to run as unpriviledged user, root is not required.")
+	}
+	http.Handle(*metricsPath, promhttp.Handler())
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
remove `/` endpoint and add `ready` or `healthz` endpoints

Reply:
done!